src/StaticAnalysis/PHPStan/Adapter/PHPStanAdapter.php

```diff
@@ -93,6 +101,42 @@ public function getVersion(): string
         return $this->version ??= $this->retrieveVersion();
     }
 
+    public function assertMinimumVersionSatisfied(): void
+    {
+        $version = $this->getVersion();
+        $majorVersion = (int) explode('.', $version)[0];
+
+        // we assume all versions greater than 2.1.17 have needed functionality
+        if ($majorVersion > self::VERSION_2) {
+            return;
+        }
+
+        if (
+            $majorVersion === self::VERSION_2
+            && (
+                version_compare($version, '2.1.17', '>=')
+                || str_starts_with($version, '2.1.x-dev') // allow dev versions for development
```

Comment:
not sure I like this line, as you might easily forget to add 2.2.x etc support as soon as PHPStan creates a new minor version

Reply:
> as you might easily forget to add 2.2.x etc support as soon as PHPStan creates a new minor version.

we don't need it, because 

- `2.2.x` is [handled by first condition](https://3v4l.org/dQYI8): `version_compare($version, '2.1.17', '>=')`
- this condition (`str_starts_with($version, '2.1.x-dev'`) is only needed to support `2.1.x` because `version_compare(`2.1.x`, '2.1.17', '>=') === false`